### PR TITLE
Fast doc build

### PR DESCRIPTION
**Pull request by Pyrsos**
Sorry, made a mess on this with the branches. Here are the latest changes for building the docs fast issue #4435
Again apologies for the mess...

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [ ] important background, or details about the implementation
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [ ] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

**Review by OriolAbril — COMMENTED** [review]
Looks good, I would name the file `fast_build_docs.sh` though, as this one does not deploy them to github pages (and should not)

File: fast_build_docs.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
```
Comment (MarcoGorelli) [inline fast_build_docs.sh]: This didn't work for me, I had to change it to:
```suggestion
#!/bin/bash
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks @Pyrsos , much faster like this!

Have just left one small comment, other than that it looks good to me

**Comment by MarcoGorelli** [comment]
As an aside, maybe there should be another script `build_docs.sh` which:
- fetches the latest changes from the `main` branch in `pymc-examples` (`git submodule update --remote`)
- builds the full docs without deploying them

Even better if they could be Python scripts instead of bash ones, so that any contributor on any OS can easily build the docs with just one script

Anyway, all this could be a separate PR

----

Last piece of advice here would be to make PRs from new branches rather than from master

**Comment by Pyrsos** [comment]
Sorry for the delay @MarcoGorelli, I have now made the suggested changes. I'd be happy to work on a separate PR for your suggestions to change from bash scripts to Python scripts. Should I open an issue for this?

**Review by MarcoGorelli — APPROVED** [review]
cool, thanks!

and yes, I think opening an issue would be good so it doesn't get forgotten

**Review by OriolAbril — APPROVED** [review]

**Comment by Pyrsos** [comment]
OK, I have opened another issue here #4483. Is there anything else I need to do with this one or can it be merged? I do not think I have write access to the repository to do the merge.

**Comment by OriolAbril** [comment]
This is all for now, the other tasks have issue remainders and can be done in follow up PRs. Thanks for the PR @Pyrsos !

**Pull request merged**